akka-actor/src/main/scala/akka/util/Helpers.scala

```diff
@@ -130,12 +130,20 @@ object Helpers {
    * INTERNAL API
    */
   private[akka] final implicit class ConfigOps(val config: Config) extends AnyVal {
+
+    import JavaDurationConverters._
+
     def getMillisDuration(path: String): FiniteDuration = getDuration(path, TimeUnit.MILLISECONDS)
 
     def getNanosDuration(path: String): FiniteDuration = getDuration(path, TimeUnit.NANOSECONDS)
 
     private def getDuration(path: String, unit: TimeUnit): FiniteDuration =
       Duration(config.getDuration(path, unit), unit)
+
+    def getPotentiallyInfiniteDuration(path: String): Duration = config.getString(path) match {
+      case "infinite" => Duration.Inf
+      case _ => config.getDuration(path).asScala
```

Comment:
should we cover `Undefined` as well, for `""` and `"off"` values?

Reply:
Not undefined, that must be an error.

Reply:
why? we use `Undefined` in some places, e.g. https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala#L89

Reply:
well, I actually think both cases belong to `getPotentiallyInfiniteDuration`.
Both `Duration.Undefined` and `Duration.Inf` are `Infinite` so the name is alright.

Reply:
```
    def getPotentiallyInfiniteDuration(path: String): Duration = toRootLowerCase(config.getString(path)) match {
         case "infinite" => Duration.Inf
         case "" | "off" => Duration.Undefined
         case _ => config.getDuration(path).asScala
     }
```